src/main/java/uk/gov/companieshouse/filinghistory/api/repository/Repository.java

```diff
@@ -102,7 +102,7 @@ public Optional<FilingHistoryDocument> findByIdAndCompanyNumber(final String id,
 
             return Optional.ofNullable(mongoTemplate.findOne(query, FilingHistoryDocument.class));
         } catch (DataAccessException ex) {
-            LOGGER.error("MongoDB error when finding the document: %s".formatted(ex.getMessage()), ex,
+            LOGGER.error("MongoDB error when finding document: %s".formatted(ex.getMessage()), ex,
```

Comment:
We need to expand the catch block into multiple catches for specific exceptions, currently its difficult to figure out what when wrong and how important it its.